datafusion/physical-expr/src/expressions/cast.rs

```diff
@@ -271,12 +271,18 @@ mod tests {
 
     #[test]
     fn test_cast_decimal_to_decimal() -> Result<()> {
-        let array = vec![1234, 2222, 3, 4000, 5000];
+        let array = vec![
```

Comment:
This was a bug in the test -- the input data had 5 rows but the expected value had 6 values (and apparently `is_valid()` used to return `false` for an index off the end of the array and in arrow 19 it returns `true`)

Reply:
cc @liu